### PR TITLE
[DataGrid] Pass reason to `onPaginationModelChange`

### PR DESCRIPTION
**Pull request by bjackson42**
Fixes https://github.com/mui/mui-x/issues/12897

- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui/mui-x/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

### CODE REVIEW

File: scripts/README.md [hunk 1]
```diff
@@ -30,7 +30,7 @@ pnpm release:changelog
 > `<!--/ DO_NOT_REMOVE /-->`
 > This string needs to stay where it gets inserted for the automated discord announcement to work.
 
-You can also provide the GitHub token by setting `process.env.GITHUB_TOKEN` variable.
+You can also provide the github token by setting `process.env.GITHUB_TOKEN` variable.
```
Comment (cherniavskii) [inline scripts/README.md:33]: Why?

File: packages/x-data-grid-premium/src/hooks/features/aggregation/gridAggregationInterfaces.ts [hunk 2]
```diff
@@ -74,7 +74,7 @@ export interface GridAggregationFunction<V = any, AV = V> {
   getCellValue?: (params: GridAggregationGetCellValueParams) => V;
 }
 
-interface GridAggregationParams<V = any> {
+export interface GridAggregationParams<V = any> {
```
Comment (cherniavskii) [inline packages/x-data-grid-premium/src/hooks/features/aggregation/gridAggregationInterfaces.ts:77]: Why?

File: packages/x-data-grid/src/hooks/features/pagination/useGridPaginationModel.ts [hunk 3]
```diff
@@ -115,17 +115,20 @@ export const useGridPaginationModel = (
...
+          pagination: {
+            ...state.pagination,
+            paginationModel: getDerivedPaginationModel(
+              state.pagination,
+              props.signature,
+              paginationModel,
+            ),
+          },
+        }),
+        'setPaginationModel',
```
Comment (cherniavskii) [inline packages/x-data-grid/src/hooks/features/pagination/useGridPaginationModel.ts:130]: Could you also pass the "stateRestorePreProcessing" reason in the `stateRestorePreProcessing` function?

**Review by cherniavskii — COMMENTED** [review]
Thank you!

File: packages/x-data-grid-premium/src/hooks/features/aggregation/gridAggregationInterfaces.ts (hunk 2 above)
Comment (cherniavskii) [reply packages/x-data-grid-premium/src/hooks/features/aggregation/gridAggregationInterfaces.ts:77]: I think you had an outdated `master` branch in your MUI X fork.
I'll merge master into it 👍🏻

**Comment by DungTiger** [comment]
@cherniavskii Seems no activity here, I just created a new PR here
https://github.com/mui/mui-x/pull/13959
Which contains the fix, your comment and a type fix.

**Comment by DungTiger** [comment]
@bjackson42 I think you need to rebase your changes on latest `next` branch. It appears you have more changes (and commits) than needed

**Comment by dhrumit9** [comment]
Hello, this looks quite stale. Any updates on getting this merged ?

**Comment by MBilalShafi** [comment]
Closing in favor of a similar one #13959

**Pull request closed without merging**